### ISSUE TITLE
Bug: `project: true` doesn't permit trailing pathsep in `tsconfigRootDir`

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#11413]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

I expect that `project: true` should search for `tsconfig.json` files in the `tsconfigRootDir`, even if it's written with a trailing slash. 

### Reproduction Repository Link

https://github.com/kirkwaiblinger/typescript-eslint-project-true-trailing-pathsep

### Repro Steps

1) clone the repo
2) `npm i`
3) `npm run lint`

### Versions

```js
{
  "eslint": "9.31.0",
  "typescript": "5.8.3",
  "typescript-eslint": "8.37.0"
}
```
